lightning/src/ln/channelmanager.rs

```diff
@@ -941,13 +941,15 @@ impl<ChanSigner: ChannelKeys, M: Deref, T: Deref, K: Deref, F: Deref, L: Deref>
 				msg: update
 			});
 		}
+
+		Ok(())
 	}
 
 	/// Force close all channels, immediately broadcasting the latest local commitment transaction
 	/// for each to the chain and rejecting new HTLCs on each.
 	pub fn force_close_all_channels(&self) {
 		for chan in self.list_channels() {
-			self.force_close_channel(&chan.channel_id);
+			self.force_close_channel(&chan.channel_id).unwrap();
```

Comment:
This is race-y - if we close a channel in another thread while this is running we'll panic needlessly.

Reply:
Rebased in 0e83723